cves/CVE-2014-3583.yml

```diff
@@ -223,4 +249,23 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+    This was a coding mistake, and to an extent, a design mistake. When Apache was
+    initially writing this function, they assumed that any header coming in from another
+    server would include the buffer end character, '\0'. This null character is the only
+    mechanism that stops C from reading memory past the end of the string. 
```

Comment:
Is the null character even part of the HTTP standard? I was under the assumption that a newline was what delimited different headers.  If it isn't it seems like a massive oversight on Apache's part.

Reply:
Not entirely sure what the under-the-hood mechanisms are to get information from servers into memory. I'll try to look into it to see if I can find anything